replace replace with regex sub

harrypuuter opened the pull request.

harrypuuter committed:
replace replace with regex sub

mburkart commented on variations.py:
```diff
@@ -98,29 +98,32 @@ def create(self, unit):
                     replaced = True
-                    logger.debug(f"Replaced act {quantity} with {act.variable}")
+                    logger.debug(f"Replaced act {old_act} with {act.variable}")
+                    old_act = deepcopy(act.variable)
```
Do we really need the deep copies here?

harrypuuter replied:
yes

mburkart committed:
Only use multiprocessing when necessary.

This helps debugging as python creates an informative stacktrace.

mburkart committed:
Allow reading of input ntuples via xrootd.

Read the list of available input files via xrootd filesystem.
Switch to static ROOT.TFile.Open as ROOT.TFile can not handle xrootd
requests.
Reset all trees after writing outputs to close all file handles.

mburkart committed:
Remove unnecessary opening of root files

mburkart committed:
Merge pull request #60 from KIT-CMS/xrd-read

Allow reading via xrootd

mburkart committed:
Merge branch 'master' into regex_replace
